plonky2x/src/frontend/vars/array.rs

```diff
@@ -162,18 +170,75 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
         V::from_variables_unsafe(&selected_vars)
     }
 
-    pub fn get_fixed_subarray<const SIZE: usize>(
+    pub fn get_fixed_subarray<const MAX_ARRAY_SIZE: usize, const SUB_ARRAY_SIZE: usize>(
         &mut self,
-        array: &[Variable],
-        start: Variable,
-    ) -> ArrayVariable<Variable, SIZE> {
-        let mut res = Vec::new();
-        for i in 0..SIZE {
-            let i_variable = self.constant(L::Field::from_canonical_usize(i));
-            let index = self.add(start, i_variable);
-            res.push(self.select_array(array, index));
+        array: &ArrayVariable<Variable, MAX_ARRAY_SIZE>,
+        array_size: Variable,
+        start_idx: Variable,
+        seed: &ArrayVariable<ByteVariable, 15>, // TODO: Seed it with 120 bits.  Need to figure out if this is secure.
```

Comment:
Regarding using ByteVariable, I think the vast majority of the time, the seed would be a hash, so a bytesvariable type.

Regarding arbitrary size, specifying the seed length in the type will help enforce that during compile time, and having the check done then is generally better than run time.

Reply:
Don't you think that people might want to seed it with headers that are 32 bytes and instead of passing in something of length 15, it's easier for them to just pass in a `Vec`? The reason I'm kind of opposed to this `15` hardcoded here is that it's a minimum--it's not always the case that it's 15. Also have `ByteVariable` here to me seems overly restrictive, it's just any sort of hash or randomness in the circuit that suffices, right? That randomness could come from Variables as well?

Reply:
I think that the risk of having Variables passed in is that they could be representing something arbitrary (e.g. they could be bits or bytes).  So its not very clear on how many bits of security the seed would be after decomposing the variable type.  Also, rust's rng seed function takes in a vector of u8's (https://rust-random.github.io/rand/rand/trait.SeedableRng.html#associatedtype.Seed).

As for the 15 hard coded limit, that's fair.